src/registry.ts

```diff
@@ -149,37 +149,37 @@ interface BtnListener {
      * Emitted when a button is pressed. Gets passed the instance of the clicked button.  
      * Make sure to check that you're responding to the correct `btn`'s interaction by validating the customId!
      */
-    on(event: "press", listener: (int: ButtonInteraction, btn: MessageButton) => void): this;
+    on(event: "press", listener: (int: ButtonInteraction, btn: ButtonBuilder) => void): this;
 }
 
 // TODO:FIXME: I'm pretty sure this causes memory leaks if the events aren't cleaned up -sv
 class BtnListener extends EventEmitter
 {
-    private btns = new Collection<string, MessageButton>();
+    private btns = new Collection<string, ButtonBuilder>();
 
     constructor()
     {
         super({ captureRejections: true });
     }
 
-    /** Adds multiple MessageButtons. Throws an error if there's no customId props! */
-    public addBtns(btns: MessageButton[]): void
-    /** Adds one MessageButton. Throws an error if there's no customId prop! */
-    public addBtns(btn: MessageButton): void
-    public addBtns(btns: MessageButton | MessageButton[]): void
+    /** Adds multiple ButtonBuilders. Throws an error if there's no customId props! */
+    public addBtns(btns: ButtonBuilder[]): void
+    /** Adds one ButtonBuilder. Throws an error if there's no customId prop! */
+    public addBtns(btn: ButtonBuilder): void
+    public addBtns(btns: ButtonBuilder | ButtonBuilder[]): void
     {
         btns = Array.isArray(btns) ? btns : [btns];
 
         btns.forEach(bt => {
-            if(!bt.customId)
-                throw new TypeError(`MessageButton "${bt.label}/${bt.style}" doesn't have a customId`);
-            this.btns.set(bt.customId, bt);
+            if(!(bt.data as Partial<APIButtonComponentWithCustomId>).custom_id)
+                throw new TypeError(`ButtonBuilder "${bt.data.label}/${bt.data.style}" doesn't have a customId`);
```

Comment:
If this displays the enum as a number, then I think it could be confused with the button index

Reply:
wdym exactly? plz elaborate